bug(ci): rollback-runner workflow false-fails because gh cli not in toolchain-rust image

## Summary

The `Rollback Runner` workflow's **`Verify target release asset exists`** step
false-fails on every run, blocking any production rollback. The error message
claims the release is missing even when it exists with the correct asset.

## Evidence

- Example failing run: [actions/runs/24655378707](https://github.com/vm0-ai/vm0/actions/runs/24655378707/job/71245...) (rolling back to `v0.86.4`, which absolutely exists with `runner-v0.86.4-aarch64-linux` attached).
- Prior failing run: [actions/runs/24627402992](https://github.com/vm0-ai/vm0/actions/runs/24627402992) (same failure mode, different day).
- Log excerpt:
  ```
  Run TAG="runner-rs-v${VERSION}"
    TAG="runner-rs-v${VERSION}"
    ASSET_NAME="runner-v${VERSION}-aarch64-linux"
    if ! gh release view "$TAG" --json assets --jq '.assets[].name' 2>/dev/null | grep -qx "$ASSET_NAME"; then
      echo "::error::Release $TAG does not exist or is missing asset $ASSET_NAME."
      exit 1
    fi
    shell: sh -e {0}
    env:
      GH_TOKEN: ***
      VERSION: 0.86.4

  Error: Release runner-rs-v0.86.4 does not exist or is missing asset runner-v0.86.4-aarch64-linux.
  Error: Process completed with exit code 1.
  ```
- The failing step runs in **52 ms** — far too fast for a real `gh` API roundtrip; it's a local failure.
- Verified locally: `gh release view runner-rs-v0.86.4 --json assets` returns
  `{"assets":["runner-v0.86.4-aarch64-linux"],"tag":"runner-rs-v0.86.4"}`.

## Root cause

The job runs in `ghcr.io/vm0-ai/vm0-toolchain-rust:20260416`
(`.github/workflows/rollback-runner.yml:53`). In `docker/toolchain/Dockerfile`,
**`gh` is installed in stage 3 (`development`), not in the stage 2
(`toolchain-rust`) that this workflow pulls** — see lines 138–143.

Three shell behaviours compound to silently mislabel the failure as "release
missing":

1. `2>/dev/null` swallows the `sh: 1: gh: not found` error that would otherwise
   pinpoint the real problem.
2. The container has no `bash` preferred by default, so GitHub falls back to
   `sh -e {0}` (POSIX `sh` with `errexit`). POSIX `sh` has **no `pipefail`**,
   so the pipeline's exit status comes from `grep` alone.
3. Empty stdin into `grep -qx` returns 1 → the `if !` branch fires → the
   misleading `::error::Release … does not exist` line is printed.

End result: any rollback attempt dies at the preflight, before the real
SSH / Ansible steps can even start.

## Why it wasn't caught

`release-please.yml` shares the same image, but its upload step
(`release-please.yml:1211–1244`) already works around the missing `gh` by
hitting the GitHub REST API with `curl` + `jq` directly — those two tools
*are* in stage 2. The rollback workflow was the only place that used `gh`
against `toolchain-rust`, and rollbacks hadn't been exercised since the
workflow landed.

## Scope of impact

**P1 / break-glass tool.** This is the workflow we reach for during prod
incidents; it's currently 100% broken at step 1. Note the concurrency
mutex (`concurrency: runner-deploy`) is taken *after* this step would fail,
so there's no lingering lock — but there's also no successful rollback path.

## Proposed fix

Swap the one failing step to `curl` + `jq` (already in the base image) and
stop silencing stderr. Mirrors the pattern `release-please.yml` already uses
when it talks to the Release API:

```yaml
- name: Verify target release asset exists
  env:
    GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    VERSION: ${{ inputs.target_version }}
  shell: bash
  run: |
    set -euo pipefail
    TAG="runner-rs-v${VERSION}"
    ASSET_NAME="runner-v${VERSION}-aarch64-linux"
    ASSETS=$(curl -sfL \
      -H "Authorization: token ${GH_TOKEN}" \
      -H "Accept: application/vnd.github+json" \
      "https://api.github.com/repos/${GITHUB_REPOSITORY}/releases/tags/${TAG}" \
      | jq -r '.assets[].name')
    if ! printf '%s\n' "$ASSETS" | grep -qx "$ASSET_NAME"; then
      echo "::error::Release $TAG does not exist or is missing asset $ASSET_NAME."
      exit 1
    fi
```

Scope: single step in `.github/workflows/rollback-runner.yml`. No other
workflows touched — `release-please.yml` already uses this pattern. No
Dockerfile changes (adding `gh` to stage 2 would bloat every Rust CI job
for this one niche use).

## Verification

- Manually dispatch `Rollback Runner` with `target_version=0.86.4`,
  `confirmed_compatible=no` — expect the workflow to pass the verify step
  and then abort at the `confirmed_compatible` gate (existing behaviour,
  confirms the asset-lookup step no longer false-fails).
- Also manually dispatch with a deliberately-bogus `target_version=9.9.9` —
  expect the verify step to correctly report the release as missing.

## Not in scope

- Auditing other workflows for similar `gh`-in-`toolchain-rust` misuse.
  Pre-check (`grep -l 'gh ' .github/workflows`) says rollback-runner.yml is
  the only offender; if that changes, open a follow-up.
- Cross-workflow rollback/deploy mutex (already tracked via the TODO at
  `rollback-runner.yml:40–44` → issue #9746).

---

_Created from conversation investigating the Rollback Runner #2 failure._

--- a/.github/workflows/rollback-runner.yml
+++ b/.github/workflows/rollback-runner.yml
@@ -115,10 +115,22 @@ jobs:
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           VERSION: ${{ inputs.target_version }}
+        shell: bash
         run: |
+          set -euo pipefail
           TAG="runner-rs-v${VERSION}"
           ASSET_NAME="runner-v${VERSION}-aarch64-linux"
-          if ! gh release view "$TAG" --json assets --jq '.assets[].name' 2>/dev/null | grep -qx "$ASSET_NAME"; then
+          # curl + jq instead of `gh` — the toolchain-rust image doesn't
+          # ship `gh` (it's in stage 3 `development`, not stage 2). Same
+          # REST endpoint and auth as the upload step in release-please.yml.
+          # `curl -f` turns HTTP 404 on a missing tag into a non-zero exit
+          # so set -e halts before grep runs.
+          ASSETS=$(curl -sfL \
+            -H "Authorization: token ${GH_TOKEN}" \
+            -H "Accept: application/vnd.github+json" \
+            "https://api.github.com/repos/${GITHUB_REPOSITORY}/releases/tags/${TAG}" \
+            | jq -r '.assets[].name')
+          if ! printf '%s\n' "$ASSETS" | grep -qx "$ASSET_NAME"; then
             echo "::error::Release $TAG does not exist or is missing asset $ASSET_NAME."
             exit 1
           fi

